src/os/strings.h

```diff
@@ -84,13 +84,13 @@
 
 	#if !defined(__MINGW32__) && defined(__GNUC__) && !defined(snprintf)
 		/* (v)snprintf is in fact C99, but we like to use it over (v)sprintf for the obvious reasons */
-		#if !defined(__APPLE__) && !defined(TOS) && !defined(__FreeBSD__) && !defined(__OpenBSD__) && !defined(__DJGPP__)
+		#if !defined(__APPLE__) && !defined(TOS) && !defined(__FreeBSD__) && !defined(__OpenBSD__) && !defined(__DJGPP__) && !defined(_XOPEN_SOURCE) && (defined(_XOPEN_SOURCE) && (_XOPEN_SOURCE < 500))
```

Comment:
there is a logical impossibility : 
`!defined(_XOPEN_SOURCE) && (defined(_XOPEN_SOURCE) && (_XOPEN_SOURCE < 500))`
Do you mean 
`(!defined(_XOPEN_SOURCE) || (defined(_XOPEN_SOURCE) && (_XOPEN_SOURCE < 500)))` ?

Reply:
The plain english version of what I attempted to code here:

* If you do not have _XOPEN_SOURCE, use the definitions below.
* If you do have _XOPEN_SOURCE and it's below 500 (value which I determined by reading strdup(3) and printf(3) on my system), use the definitions below.

I think that your implementation fits this description, mine doesn't after carefully reviewing this in my head. I also found why it was working on my machine, the related change I've done to "config.lib" is incomplete and running:

```
make VERBOSE=1
```

did reveal that I didn't have "-D_XOPEN_SOURCE=500" like I was expecting. I will have to fix this mess...